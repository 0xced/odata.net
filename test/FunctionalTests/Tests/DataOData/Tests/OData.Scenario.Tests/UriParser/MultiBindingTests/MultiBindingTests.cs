using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;
using Microsoft.OData.Edm;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Microsoft.Test.Taupo.OData.Scenario.Tests.UriParser.MultiBindingTests
{
    [TestClass]
    public class MultiBindingTests : UriParserTestsBase
    {
        private readonly Uri entitySetBase = new Uri("http://host/EntitySet");
        private readonly Uri complexProp1Base = new Uri("http://host/EntitySet('abc')/complexProp1");
        private readonly Uri navUnderComplexBase = new Uri("http://host/EntitySet('abc')/complexProp1/CollectionOfNavOnComplex");
        private readonly Uri containedNav1Base = new Uri("http://host/EntitySet('abc')/ContainedNav1");
        private readonly IEdmModel bindigModel = MultiBindingModel.GetModel();

        [TestMethod]
        [MethodImpl(MethodImplOptions.NoOptimization)]
        public void NestedExpandUnderComplex()
        {
            // http://host/EntitySet?$expand=complexProp1/CollectionOfNavOnComplex($expand=NavNested)
            this.ApprovalVerifyExpandParser(entitySetBase, "complexProp1/CollectionOfNavOnComplex($expand=NavNested)", bindigModel);
            this.TestAllInOneExtensionExpand(entitySetBase, "complexProp1/collectionofnavoncomplex($expand=NavNested)", "complexProp1/CollectionOfNavOnComplex($expand=NavNested)", bindigModel);
        }

        [TestMethod]
        [MethodImpl(MethodImplOptions.NoOptimization)]
        public void NestedExpandUnderSplitComplex()
        {
            // http://host/EntitySet('abc')/complexProp1?$expand=CollectionOfNavOnComplex($expand=NavNested)
            this.ApprovalVerifyExpandParser(complexProp1Base, "CollectionOfNavOnComplex($expand=NavNested)", bindigModel);
            this.TestAllInOneExtensionExpand(complexProp1Base, "CollectionOfNavOnComplex($expand=navnested)", "CollectionOfNavOnComplex($expand=NavNested)", bindigModel);
        }

        [TestMethod]
        [MethodImpl(MethodImplOptions.NoOptimization)]
        public void ExpandNested()
        {
            // http://host/EntitySet('abc')/complexProp1/CollectionOfNavOnComplex?$expand=NavNested
            this.ApprovalVerifyExpandParser(navUnderComplexBase, "NavNested", bindigModel);
            this.TestAllInOneExtensionExpand(navUnderComplexBase, "navNested", "NavNested", bindigModel);
        }

        [TestMethod]
        [MethodImpl(MethodImplOptions.NoOptimization)]
        public void OrderByContained()
        {
            // http://host/EntitySet('abc')?$orderby=ContainedNav1/NavOnContained/ID
            this.ApprovalVerifyOrderByParser(entitySetBase, "ContainedNav1/NavOnContained/ID", bindigModel);
            this.TestAllInOneExtensionOrderBy(entitySetBase, "containednav1/NavOnContained/ID", "ContainedNav1/NavOnContained/ID", bindigModel);
        }

        [TestMethod]
        [MethodImpl(MethodImplOptions.NoOptimization)]
        public void FilterContained()
        {
            // http://host/EntitySet('abc')/ContainedNav1?$filter=NavOnContained/ID eq 'abc'
            this.ApprovalVerifyFilterParser(containedNav1Base, "NavOnContained/ID eq 'abc'", bindigModel);
            this.TestAllInOneExtensionFilter(containedNav1Base, "navOnContained/ID eq 'abc'", "NavOnContained/ID eq 'abc'", bindigModel);
        }
    }
}
