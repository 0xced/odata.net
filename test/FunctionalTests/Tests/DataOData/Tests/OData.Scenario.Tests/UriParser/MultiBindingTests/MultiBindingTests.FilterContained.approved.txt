$filter = NavOnContained/ID eq 'abc'


FilterQueryOption
	ItemType = [NS.ContainedEntityType Nullable=False]
	Parameter = 
	EntityRangeVariable
		Name = $it
		NavigationSource = ContainedNav1
		TypeReference = [NS.ContainedEntityType Nullable=False]
	Expression = 
		ConvertNode
			TypeReference = [Edm.String Nullable=True Unicode=True]
			Source = 
			SingleValuePropertyAccessNode
				Property = ID
				TypeReference = [Edm.String Nullable=False Unicode=True]
				Source = 
				SingleNavigationNode
					NavigationSource = NavEntitySet1
					Type Reference = [NS.NavEntityType Nullable=False]
					Property = NavOnContained
					Multiplicity = One
					Source = 
					EntityRangeVariableReferenceNode
						Name = $it
						NavigationSource = ContainedNav1
						TypeReference = [NS.ContainedEntityType Nullable=False]
						Range Variable = Microsoft.OData.UriParser.EntityRangeVariable
	Equal
		abc(Type: [Edm.String Nullable=True Unicode=True])